app/services/api/ApiCall.tsx

```diff
@@ -106,7 +106,7 @@ class ApiCall {
    * Makes the fetch request and awaits the response or error. Also handles the redirect to error
    * or login pages
    */
-  async request(confirm = false, ignoreAbort = false): Promise<RequestReturnType> {
+  async request(confirm = false, ignoreAbort = false) {
```

Comment:
The actual return type is controlled by the `raw` variable, but if I define it using `type1 | type2` here then the code that uses this will show errors about the type. Lmk if that makes sense

Reply:
It's definitely a larger problem and maybe a bit off topic from this PR. Part of the issue is that 95% of responses will return `Record<string, unknown>[] | null (if it errors)` and 5% will return `ResponseType` if `raw` is passed to the function.

I could make a ticket for it though it won't be very high priority to cast all the API call responses

Reply:
DEVSU-1607